sql/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -3723,7 +3728,47 @@ class SparkConnectPlanner(
     LazyExpression(transformExpression(getLazyExpression.getChild))
   }
 
-  private def assertPlan(assertion: Boolean, message: String = ""): Unit = {
+  private def transformSubqueryExpression(
+      getSubqueryExpression: proto.SubqueryExpression): Expression = {
+    val planId = getSubqueryExpression.getPlanId
+    getSubqueryExpression.getSubqueryType match {
+      case proto.SubqueryExpression.SubqueryType.SUBQUERY_TYPE_SCALAR =>
+        UnresolvedScalarSubqueryPlanId(planId)
+      case proto.SubqueryExpression.SubqueryType.SUBQUERY_TYPE_EXISTS =>
+        UnresolvedExistsPlanId(planId)
+      case other => throw InvalidPlanInput(s"Unknown SubqueryType $other")
+    }
+  }
+
+  private def transformWithRelations(getWithRelations: proto.WithRelations): LogicalPlan = {
+    // Wrap the plan to keep the original planId.
+    val plan = Project(Seq(UnresolvedStar(None)), transformRelation(getWithRelations.getRoot))
```

Comment:
I am not sure I understand why you need to wrap this.

Reply:
Usually the plan returned by `transformRelation(getWithRelations.getRoot)` already has a plan id, but it will be overwritten after this `transformWithRelations` called from another `transformRelation`, then plan id matching, e.g., for resolution of `df[col]`, may go wrong.
So we need to return a different plan from `transformWithRelations`.